src/Microsoft.Extensions.SecretManager.Tools/Internal/SetCommand.cs

```diff
@@ -34,72 +31,74 @@ public static void Configure(CommandLineApplication command, CommandLineOptions
 
             command.OnExecute(() =>
             {
-                options.Command = new SetCommand(nameArg.Value, valueArg.Value);
-            });
-        }
+                if (console.IsInputRedirected && nameArg.Value == null)
+                {
+                    options.Command = new FromStdInStrategy();
+                    return 0;
+                }
 
-        internal SetCommand(string keyName, string keyValue)
-        {
-            Debug.Assert(keyName != null || keyValue == null, "Inconsistent state. keyValue must not be null if keyName is null.");
-            _keyName = keyName;
-            _keyValue = keyValue;
-        }
+                if (string.IsNullOrEmpty(nameArg.Value))
+                {
+                    console.Error.WriteLine(Resources.FormatError_MissingArgument("name").Red());
+                    return 1;
+                }
 
-        internal SetCommand()
-        { }
+                if (valueArg.Value == null)
+                {
+                    console.Error.WriteLine(Resources.FormatError_MissingArgument("value").Red());
+                    return 1;
+                }
 
-        public void Execute(CommandContext context)
-        {
-            if (context.Console.IsInputRedirected && _keyName == null)
-            {
-                ReadFromInput(context);
-            }
-            else
-            {
-                SetFromArguments(context);
-            }
+                options.Command = new ForOneValueStrategy(nameArg.Value, valueArg.Value);
+                return 0;
+            });
         }
 
-        private void ReadFromInput(CommandContext context)
+        public class FromStdInStrategy : ICommand
```

Comment:
Open to naming suggesions. Alternatives I considered:

ConsoleInputSetCommand
ConsoleInputStrategy
ICanHazPipedInputCommand